### DISCUSSION TITLE
Interested in Sway Service?

### DISCUSSION THREAD

**Discussion by ldelossa** [Aylur/ags#202] [Q&A] [answered]
I'm playing with AGS now. 

Would you be open to a Sway service similar to the Hyprland one as a contribution from me?

See any issues with attempting this?

**Comment by Aylur** [comment, marked as answer]
Of course. I know @kotontrion has started working on it, but I don't know if there are plans to finish it.
https://github.com/Aylur/ags/blob/c89da02cbc646a75b18331cc3a13b3d0c2d57500/src/service/sway.ts

**Reply by kotontrion** [reply]
I halted the development of this, as there was basically no interest for it. But it could be picked up again. I'm not quite sure where I left off, but I think I got the IPC working, so you could use my branch as your base. If you any need help with this, let me know.

**Reply by ldelossa** [reply]
Great @kotontrion  id love to use your base work here. 

I need to implement the Workspaces component from Aylur's dotfiles soon, (got just about everything else working). 

When i start this I'll fork ags and use your base to implement a Sway service. 

Thanks all.

**Reply by ldelossa** [reply]
Hey @Aylur @kotontrion 

I'm trying to dev `ags` in a container.
I have things building, however when I go to run ags it fails to load any widgets in the mounted in `~/.config/ags` directory.

I know its not the config, since the config works on my host system with `ags` installed from copr. Any ideas why this may occur? `ags` basically isn't loading the config files, or at the minimum not loading my widows.

**Reply by ozwaldorf** [reply]
Any progress on this?

**Comment by ozwaldorf** [comment]
I was able to cherry-pick @kotontrion's work onto main and hack things to a working state: https://github.com/ozwaldorf/ags/commit/f5f0d906914521cbb0456d50c8bbf26b2f16e62f
https://github.com/ozwaldorf/ags/tree/feature/sway

**Reply by Freemason-12** [reply]
if so, why not open a PR to merge your changes?

**Reply by ozwaldorf** [reply]
> if so, why not open a PR to merge your changes?

There's an open PR already, I just haven't had much time to get back to cleaning up the last couple outstanding bugs
